binders/kafka-binder/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -1548,6 +1575,32 @@ public boolean isRunning() {
 			return this.running;
 		}
 
+		@Override
+		public void handleMessage(Message<?> message) {
+
+			// if we use our own partition handler to update partition count we recalculate partition
+			if (kafkaPartitionHandler != null) {
+				kafkaPartitionHandler.setPartitionCount(getKafkaTemplate().partitionsFor(this.topic).size());
```

Comment:
This is nice. I wasn't thinking yesterday that we could do this. Much simpler conceptually now.